src/vs/workbench/browser/layout.ts

```diff
@@ -278,14 +279,25 @@ export abstract class Layout extends Disposable implements IWorkbenchLayoutServi
 			}
 		};
 
+		// If hidden group becomes active, show all groups
+		const showEditorGroupsIfHidden = () => {
```

Comment:
I would have this within the `EditorPart` and not here in `layout.ts`. The one above for `showEditorIfHidden` makes sense to have in `layout` because its about the entire `EditorPart` which the `layout.ts` is responsible for.